edgexfoundry/edgex-compose
# feat!: Switch from Kong proxy to NGINIX proxy

BREAKING CHANGE: EdgeX 3.0 replaces the Kong reverse proxy with a generic NGINX reverse proxy.  While the usage model remains the same, there is a minor change to the method by which a JWT authentication token is obtained.

This commit removes Kong and Postgres and replaces with a much lighter weight NGINX and a new security-proxy-auth service.

## PR Checklist
Please check if your PR fulfills the following requirements:

<https://github.com/edgexfoundry/edgex-docs/pull/981>


## Testing Instructions
From edgex-compose, compose-builder directory:

```shell
$ make pull ds-virtual
$ make run ds-virtual
$ watch docker ps -a
```

Ensure that all the services are stable (stay up for 2+ minutes) then try edgex-ui:

```shell
$ make get-token
```

Then browse to http://localhost:4000, paste in the token, and go to Data Center and watch for events (this will ensure that API gateway authentication is working properly.)
